### PR TITLE
ENH: add relate() function

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: pygeos/test/test_predicates.py [hunk 1]
```diff
@@ -90,3 +90,11 @@ def test_equals_exact_tolerance():
     # default value for tolerance
     assert pygeos.equals_exact(p1, p1).item() is True
     assert pygeos.equals_exact(p1, p2).item() is False
+
+
+def test_relate():
```
Comment (caspervdw) [inline pygeos/test/test_predicates.py:95]: Can you also test with `None` input?

Comment (jorisvandenbossche) [reply pygeos/test/test_predicates.py:95]: Ah, good you ask, because I missed handling the second input for None

**Review by caspervdw — APPROVED** [review]
Feel free to merge after adding a line to the changes!

**Comment by jorisvandenbossche** [comment]
> after adding a line to the changes

which line exactly?

**Comment by caspervdw** [comment]
> > after adding a line to the changes
> 
> which line exactly?

Sorry, I meant CHANGELOG.

**Comment by jorisvandenbossche** [comment]
Done!
